Sources/Action/UIKitExtensions/UIButton+Rx.swift

```diff
@@ -51,5 +51,52 @@ public extension Reactive where Base: UIButton {
             }
         }
     }
+    
+    /// Binds enabled state of action to button, and subscribes to rx_tap to execute action with given input transform.
+    /// These subscriptions are managed in a private, inaccessible dispose bag. To cancel
+    /// them, call bindToAction with another action or nil.
+    public func bindToAction <Input,Output>(_ action:Action<Input,Output>?,_ inputTransform: @escaping (Base) -> (Input))   {
+        // This effectively disposes of any existing subscriptions.
+        self.base.resetActionDisposeBag()
+        
+        
+        // If no action is provided, there is nothing left to do. All previous subscriptions are disposed.
+        if (action == nil) {
+            return
+        }
+        // Technically, this file is only included on tv/iOS platforms,
+        // so this optional will never be nil. But let's be safe 😉
+        let lookupControlEvent: ControlEvent<Void>?
+        
+        #if os(tvOS)
+            lookupControlEvent = self.primaryAction
+        #elseif os(iOS)
+            lookupControlEvent = self.tap
+        #endif
+        
+        guard let controlEvent = lookupControlEvent else {
+            return
+        }
+        // For each tap event, use the inputTransform closure to provide an Input value to the action
+        controlEvent
+            .map { return inputTransform(self.base) }
+            .bindTo(action!.inputs)
+            .addDisposableTo(self.base.actionDisposeBag)
+        
+        // Bind the enabled state of the control to the enabled state of the action
+        action!
+            .enabled
+            .bindTo(self.isEnabled)
+            .addDisposableTo(self.base.actionDisposeBag)
+        
+    }
```

Comment:
I'll dig more deeply into this to avoid duplication, I've already tried something but got stuck.